// asposewordscloud/dist/model/bookmarkInsert.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="bookmarkInsert.ts">
 *   Copyright (c) 2023 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookmarkInsert = exports.importsMapBookmarkInsert = void 0;
const bookmarkData_1 = require("./bookmarkData");
const documentPosition_1 = require("./documentPosition");
exports.importsMapBookmarkInsert = {
    BookmarkData: bookmarkData_1.BookmarkData,
    DocumentPosition: documentPosition_1.DocumentPosition,
};
/**
 * Represents a bookmark to insert.
 */
class BookmarkInsert extends bookmarkData_1.BookmarkData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return super.getAttributeTypeMap().concat(BookmarkInsert.attributeTypeMap);
    }
    constructor(init) {
        super(init);
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
}
exports.BookmarkInsert = BookmarkInsert;
/**
 * Attribute type map
 */
BookmarkInsert.attributeTypeMap = [
    {
        name: "endRange",
        baseName: "EndRange",
        type: "DocumentPosition",
    },
    {
        name: "startRange",
        baseName: "StartRange",
        type: "DocumentPosition",
    }
];
