// asposewordscloud/dist/model/compareData.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="compareData.ts">
 *   Copyright (c) 2023 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompareData = exports.importsMapCompareData = void 0;
const compareOptions_1 = require("./compareOptions");
exports.importsMapCompareData = {
    CompareOptions: compareOptions_1.CompareOptions,
};
/**
 * Container class for compare documents.
 */
class CompareData {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return CompareData.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
}
exports.CompareData = CompareData;
/**
 * Attribute type map
 */
CompareData.attributeTypeMap = [
    {
        name: "author",
        baseName: "Author",
        type: "string",
    },
    {
        name: "compareOptions",
        baseName: "CompareOptions",
        type: "CompareOptions",
    },
    {
        name: "comparingWithDocument",
        baseName: "ComparingWithDocument",
        type: "string",
    },
    {
        name: "dateTime",
        baseName: "DateTime",
        type: "Date",
    },
    {
        name: "resultDocumentFormat",
        baseName: "ResultDocumentFormat",
        type: "string",
    }
];
