// asposewordscloud/dist/model/fileReference.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="fileReference.ts">
 *   Copyright (c) 2023 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileReference = exports.importsMapFileReference = void 0;
const uuid_1 = require("uuid");
exports.importsMapFileReference = {};
class FileReference {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return FileReference.attributeTypeMap;
    }
    constructor(source, reference, content) {
        this.source = source;
        this.reference = reference;
        this.content = content;
    }
    collectFilesContent(_resultFilesContent) {
        if (this.source == FileReference.SourceEnum.Request) {
            _resultFilesContent.push(this);
        }
    }
    getSource() {
        return this.source;
    }
    getReference() {
        return this.reference;
    }
    getContent() {
        return this.content;
    }
    static fromRemoteFilePath(remoteFilePath) {
        return new FileReference(FileReference.SourceEnum.Storage, remoteFilePath, null);
    }
    static fromLocalFileContent(localFileContent) {
        return new FileReference(FileReference.SourceEnum.Request, (0, uuid_1.v4)(), localFileContent);
    }
}
exports.FileReference = FileReference;
/**
 * Attribute type map
 */
FileReference.attributeTypeMap = [
    {
        name: "source",
        baseName: "Source",
        type: "FileReference.SourceEnum",
    },
    {
        name: "reference",
        baseName: "Reference",
        type: "string",
    }
];
(function (FileReference) {
    let SourceEnum;
    (function (SourceEnum) {
        SourceEnum[SourceEnum["Storage"] = 'Storage'] = "Storage";
        SourceEnum[SourceEnum["Request"] = 'Request'] = "Request";
    })(SourceEnum = FileReference.SourceEnum || (FileReference.SourceEnum = {}));
})(FileReference = exports.FileReference || (exports.FileReference = {}));
