// asposewordscloud/dist/model/replaceRangeDto.js
"use strict";
/*
 * --------------------------------------------------------------------------------
 * <copyright company="Aspose" file="replaceRangeDto.ts">
 *   Copyright (c) 2023 Aspose.Words for Cloud
 * </copyright>
 * <summary>
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *  of this software and associated documentation files (the "Software"), to deal
 *  in the Software without restriction, including without limitation the rights
 *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *  copies of the Software, and to permit persons to whom the Software is
 *  furnished to do so, subject to the following conditions:
 *
 *  The above copyright notice and this permission notice shall be included in all
 *  copies or substantial portions of the Software.
 *
 *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 *  SOFTWARE.
 * </summary>
 * --------------------------------------------------------------------------------
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplaceRangeDto = exports.importsMapReplaceRangeDto = void 0;
exports.importsMapReplaceRangeDto = {};
/**
 * DTO container with a range element.
 */
class ReplaceRangeDto {
    /**
     * Returns attribute type map
     */
    static getAttributeTypeMap() {
        return ReplaceRangeDto.attributeTypeMap;
    }
    constructor(init) {
        Object.assign(this, init);
    }
    collectFilesContent(_resultFilesContent) {
    }
}
exports.ReplaceRangeDto = ReplaceRangeDto;
/**
 * Attribute type map
 */
ReplaceRangeDto.attributeTypeMap = [
    {
        name: "text",
        baseName: "Text",
        type: "string",
    },
    {
        name: "textType",
        baseName: "TextType",
        type: "ReplaceRangeDto.TextTypeEnum",
    }
];
/**
 * Enums for ReplaceRangeDto
 */
// tslint:disable:quotemark
// tslint:disable-next-line:no-namespace
(function (ReplaceRangeDto) {
    let TextTypeEnum;
    (function (TextTypeEnum) {
        TextTypeEnum[TextTypeEnum["Text"] = 'Text'] = "Text";
        TextTypeEnum[TextTypeEnum["Html"] = 'Html'] = "Html";
    })(TextTypeEnum = ReplaceRangeDto.TextTypeEnum || (ReplaceRangeDto.TextTypeEnum = {}));
})(ReplaceRangeDto = exports.ReplaceRangeDto || (exports.ReplaceRangeDto = {}));
// tslint:enable:quotemark
